tests/5.0/declare_target/test_declare_target_device_type_any.F90

```diff
@@ -0,0 +1,82 @@
+!===--- test_declare_target_device_type_any.F90 -----------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! The declare target directive specifies that variables, functions(C,C++ and Fortran),
+! and subroutines (Fortran) are mapped to a device. If a device_type
+! clause is present on the contained declare target directive, then its argument 
+! determines which versions are made available. When device_type(any) is specified 
+! both device and host versions of the procedure are made available.
+!
+!//===----------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+
+MODULE m_dat
+    !$omp declare target to(a, b, c, i)
+    INTEGER, DIMENSION(N) :: a, b, c
+    INTEGER :: i = 0
+END MODULE m_dat
+
+PROGRAM test_declare_target_device_type_any
+
+    USE iso_fortran_env
+    USE ompvv_lib
+    USE omp_lib
+    USE m_dat
+    implicit none
+    INTEGER :: errors = 0
+    
+    !$omp declare target to(update) device_type(any)
```

Comment:
XL 16.1.1-10 Fortran compiler fails with the following error message: 
"/gpfs/alpine/csc383/scratch/f6l/sollve/tests/5.0/declare_target/test_declare_target_device_type_any.F90", line 32.29: 1514-480 (S) The entity update that appears in the DECLARE TARGET directive must be declared in the same scoping unit.
Does it mean that we should add a declaration of the internal procedure to be used for `declare target` directive? 
Or is it a bug of the XL Fortran compiler? (Other tested OpenMP Fortran compilers (GCC, NVHPC) do not complain this.)

Reply:
OpenMP V5.0, Section 2.12.7 declare target Directve, page 183, line 24~26 says the following (OpenMP V5.2 also has the same texts):

> If a declare target directive has a device_type clause then any enclosed internal procedures cannot contain any declare target directives. The enclosing device_type clause implicitly applies to internal procedures.

This rule applies to our case since `update` is an internal procedure, which should not contain any declare target directive because of `!$omp declare target to(update) device_type(any)` in the main program unit.

Reply:
Cf. discussion in Issues #618/#619 about this. I believe the latter does not apply and the '!$omp declare target to(update) device_type(any)' should be inside 'update' itself, following the OpenMP 5.1 bug fix/clarification that added a **that** for the **to**/**enter** clause (→ aforementioned issues).

Reply:
Moved `!$omp declare target to(update) device_type(any)` to `update`, as suggested.